app/controllers/middleware_server_controller.rb

```diff
@@ -273,22 +292,26 @@ def run_server_operation(operation_info, items)
   end
 
   def trigger_mw_operation(operation, mw_server, params = nil)
-    mw_manager = mw_server.ext_management_system
-    path = mw_server.ems_ref
+    if operation == :generate_jdr
+      mw_server.enqueue_jdr_report(:requesting_user => current_userid)
+    else
+      mw_manager = mw_server.ext_management_system
+      path = mw_server.ems_ref
 
-    # in domain mode case we want to run the operation on the server-config DMR resource
-    if mw_server.respond_to?(:in_domain?) && mw_server.in_domain?
-      path = path.sub(/%2Fserver%3D/, '%2Fserver-config%3D')
-    end
+      # in domain mode case we want to run the operation on the server-config DMR resource
+      if mw_server.respond_to?(:in_domain?) && mw_server.in_domain?
+        path = path.sub(/%2Fserver%3D/, '%2Fserver-config%3D')
+      end
 
-    op = mw_manager.public_method operation
+      op = mw_manager.public_method(operation)
```

Comment:
This is slow and unadvised because it has to extract the Proc from the `BoundMethod` and then calling it. Going with `#public_send` makes more sense.